aws/resource_aws_cognito_user_pool_client_test.go

```diff
@@ -35,6 +35,68 @@ func TestAccAWSCognitoUserPoolClient_basic(t *testing.T) {
 	})
 }
 
+func TestAccAWSCognitoUserPoolClient_importBasic(t *testing.T) {
+	userPoolName := fmt.Sprintf("tf-acc-cognito-user-pool-%s", acctest.RandString(7))
+	clientName := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+
+	resourceName := "aws_cognito_user_pool_client.client"
+
+	getStateId := func(*terraform.State) (string, error) {
```

Comment:
We could shorten this lookup function if we set `*client = ...` in `testAccCheckAWSCognitoUserPoolClientExists()` from the return value of `DescribeUserPoolClient()` 

Then we can simply grab the right IDs from an object we've already fetched in there 👍

Reply:
Thanks for input. Not sure if I get your point exactly.

I've mostly copied the function and changed return values. Instead of parsing return value from `DescribeUserPoolClient()`, I directly return `userPoolId` and `clientId` set in `DescribeUserPoolClientInput` since there is error checking on `DescribeUserPoolClient()`. I assume that if there is no error then both IDs should be considered as valid.